client/asset/eth/multirpc.go

```diff
@@ -0,0 +1,1298 @@
+				_, err := p.ec.HeaderByNumber(ctx, nil /* latest */)
+				return err
+			},
+		},
+		{
+			name: "TransactionReceipt",
+			f: func(p *provider) error {
+				_, err := p.ec.TransactionReceipt(ctx, mainnetTxHash)
+				return err
+			},
+		},
+		{
+			name: "PendingNonceAt",
+			f: func(p *provider) error {
+				_, err := p.ec.PendingNonceAt(ctx, mainnetAddr)
+				return err
+			},
+		},
+		{
+			name: "SuggestGasTipCap",
+			f: func(p *provider) error {
+				tipCap, err := p.ec.SuggestGasTipCap(ctx)
+				if err != nil {
+					return err
+				}
+				log.Info("#### Retreived tip cap:", tipCap)
+				return nil
+			},
+		},
+		{
+			name: "BalanceAt",
+			f: func(p *provider) error {
+				bal, err := p.ec.BalanceAt(ctx, mainnetAddr, nil)
+				if err != nil {
+					return err
+				}
+				log.Infof("#### Vitalik's balance retrieved: %.9f", float64(dexeth.WeiToGwei(bal))/1e9)
+				return nil
+			},
+		},
+		{
+			name: "CodeAt",
+			f: func(p *provider) error {
+				code, err := p.ec.CodeAt(ctx, mainnetUSDC, nil)
+				if err != nil {
+					return err
+				}
+				log.Infof("#### %d bytes of USDC contract retrieved", len(code))
+				return nil
+			},
+		},
+		{
+			name: "CallContract(balanceOf)",
+			f: func(p *provider) error {
+				caller, err := erc20.NewIERC20(mainnetUSDC, cb)
+				if err != nil {
+					return err
+				}
+				bal, err := caller.BalanceOf(&bind.CallOpts{
+					From:    mainnetAddr,
+					Context: ctx,
+				}, mainnetAddr)
+				if err != nil {
+					return err
+				}
+				// I guess we would need to unpack the results. I don't really
+				// know how to interpret these, but I'm really just looking for
+				// a request error.
+				log.Info("#### USDC balanceOf result:", bal, "wei")
+				return nil
+			},
+		},
+		{
+			name: "ChainID",
+			f: func(p *provider) error {
+				chainID, err := p.ec.ChainID(ctx)
+				if err != nil {
+					return err
+				}
+				log.Infof("#### Chain ID: %d", chainID)
+				return nil
+			},
+		},
+		{
+			name: "PendingNonceAt",
+			f: func(p *provider) error {
+				n, err := p.ec.PendingNonceAt(ctx, mainnetAddr)
+				if err != nil {
+					return err
+				}
+				log.Infof("#### Pending nonce: %d", n)
+				return nil
+			},
+		},
+		{
+			name: "getRPCTransaction",
+			f: func(p *provider) error {
+				rpcTx, err := getRPCTransaction(ctx, p, mainnetTxHash)
+				if err != nil {
+					return err
+				}
+				var h string
+				if rpcTx.BlockNumber != nil {
+					h = *rpcTx.BlockNumber
+				}
+				log.Infof("#### RPC Tx is nil? %t, block number: %q", rpcTx.tx == nil, h)
+				return nil
+			},
+		},
+	}
+}
+
+func domain(host string) string {
+	parts := strings.Split(host, ".")
+	n := len(parts)
+	if n <= 2 {
+		return host
+	}
+	return parts[n-2] + "." + parts[n-1]
+}
+
+func checkProvidersCompliance(ctx context.Context, walletDir string, providers []*provider, log dex.Logger) error {
+	var compliantProviders map[string]bool
+	path := filepath.Join(walletDir, "compliant-providers.json")
```

Comment:
Can you explain the purpose of "compliant-providers.json" somewhere?
This is for providers that are not known in the hard coded lists, and thus go through a compliance check during wallet bring up.  And if `checkProvidersCompliance` finds it in the json file, that means it has already passed the test and to skip running the compatibility tests each time?  Will you add a doc somewhere to that effect?